web_src/js/features/clipboard.js

```diff
@@ -28,8 +58,17 @@ export default async function initClipboard() {
       if (!text) return;
 
       try {
-        await navigator.clipboard.writeText(text);
-        onSuccess(btn);
+        if (navigator.clipboard && window.isSecureContext) {
```

Comment:
Wouldn't `navigator.clipboard.writeText` throw an error in case this `clipboard-write` is being denied? It would be better to catch and handle such a error if it exists.

I think I recall copy works on non-secure origin `http://localhost`, so I think `isSecureContext` is not the only condition that a browser checks.

Reply:
It checks the "clipboard-write" and "clipboard-read" permissions, however, `navigator.permissions.query({name:"clipboard-write"})` isn't fully supported on all browsers yet.

On both firefox and chrome, the `navigator.clipboard` object only exists if the "clipboard-write" permission is granted.
It's just that on https sites, the clipboard permission is granted (for desktop at least) by default.
Also, `window.isSecureContext` is true on localhost.

The easiest way to resolve this confusion is to move the `fallbackCopyToClipboard` into the `catch {  }`, as then any error would attempt the fallback.
(so I'll do that for next commit)